### ISSUE TITLE
apps.plugin: no permission to read kthreadd or khelper io in openvz container

### ISSUE THREAD

**Issue by Splo0sh** [firehol/netdata#779]
First of all, awesome tool you created here! I instantly fell in love! : D

I just encountered a little hiccup. When the the apps_plugin is traversing the proc tree it throws an error in my virtual openvz containers.
From the /var/log/netdata/error.log:

> ERROR: apps.plugin: Cannot process /proc/2/io (command 'kthreadd/101') (errno 13, Permission denied)
> ERROR: apps.plugin: Cannot process /proc/3/io (command 'khelper/101') (errno 13, Permission denied)

Inside the container:
```
VE101# l /proc/2/io
-r-------- 1 root root 0 Aug  8 10:44 /proc/2/io
VE101# cat /proc/2/io
cat: /proc/2/io: Permission denied
```

On the physical server/Host:
```
VEHOST# ps -e |grep "kthreadd/101"
   3167 ?        00:00:00 kthreadd/101
VEHOST# l /proc/3167/io
-r-------- 1 root root 0 Apr  7 11:04 /proc/3167/io
VEHOST# cat /proc/3167/io
rchar: 0
wchar: 0
syscr: 0
syscw: 0
read_bytes: 0
write_bytes: 0
cancelled_write_bytes: 0
```

It would be nice to have an "exclude" function of process names and/or subcategories of them in the /etc/netdata/apps_groups.conf for example. Like in my case add a line containing

> exclude: kthreadd khelper

to not check them at all or something like

> kernel: kthreadd!io kauditd lockd khelper!io kdevtmpfs khungtaskd rpciod fsnotify_mark kthrotld iscsi_eh deferwq

to only ignore the not working io check. Or add

> ignore.io: kthreadd khelpe

 and parse that into an ignore list, I don't know :)

Everything else works fine, but the error.log is unnecessarily filled with a lot of errors .

Systeminfo:
Main server runs proxmox-ve-2.6.32: 3.2-129 (running kernel: 2.6.32-30-pve)
The openvz containers run Debian 7.11

Here is an extended list of permissions inside the proc from both the container and the host machine:
[netdata_apps_plugin_openvz_extended_report.txt](https://github.com/firehol/netdata/files/416963/netdata_apps_plugin_openvz_extended_report.txt)

Keep up the great work!

**Comment by ktsaou** [comment]
Nice you like it. Thanks!

I did this: netdata will log each repeating error, only once per process. So, you will get errors about these processes, but only once.

I have made a lot of changes to the code base of netdata to support health monitoring, so this will be merged with it.

**Comment by paulfantom** [comment]
Couldn't solution to this be connected with #744?

**Comment by Splo0sh** [comment]
Perfect. Thank you very much. I +1 @paulfantom's post but since #744 is still open I will go ahead and close this issue, ok?! Cheers : )
